src/record/record.js

```diff
@@ -449,7 +446,16 @@ Record.prototype._applyUpdate = function( message ) {
  */
 Record.prototype._onRead = function( message ) {
 	this.version = parseInt( message.data[ 1 ], 10 );
-	this._applyChange( jsonPath.set( this._$data, undefined, JSON.parse( message.data[ 2 ] ) ) );
+
+	this._$data = JSON.parse( message.data[ 2 ] );
+	if ( this._eventEmitter._callbacks ) {
+		var paths = Object.keys( this._eventEmitter._callbacks );
+
+		for ( var i = 0; i < paths.length; i++ ) {
+			this._eventEmitter.emit( paths[ i ], this.get( paths[ i ] ) );
```

Comment:
This should be:
```javascript
``` if ( jsonPath.get( this._$data, paths[ i ], false) !== undefined)
```